### ISSUE TITLE
VRF no-load outdoor air flow does not appear to be working correctly

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#4996]
Helpdesk ticket 10590
User:
I am modeling a VRF system for a very simple office building with hard set supply air and outdoor air control, where Cooling/Heaing Supply Air Flow Rate sets as 0.111 m3/s, No Cooling/Heating Supply Air Flow Rate sets as 0.056 m3/s; while Cooling/Heating Outdoor Air Flow Rate is 0.028 m3/s and No Load Outdoor Air Flow Rate sets as 0.014 m3/s.

After running simulation, I expect to see 0.0168 kg/s (equivalent to 0.014 m3/s) outdoor air output when there is no cooling (no heating either), but instead I see a constant 0.0336 kg/s (equivalent to 0.028m3/s). However supply air flow rate output looks fine as it switches between minimum 0.0672 kg/s (0.056 m3/s) to maximum 0.133 kg/s (0.111 m3/s) responding to whether or not the cooling power is consumed.

**Comment by Nigusse** [comment]
@mjwitte @rraustad Current code doesn't set No Load OA flow correctly for fan type Fan:ConstantVolume. Average air mass flow rate calculation depending fan operating modes is modified as follows:

```
    if ( VRFTU( VRFTUNum ).OpMode == CycFanCycCoil ) {
        AverageUnitMassFlow = ( PartLoadRatio * CompOnMassFlow ) + ( ( 1 - PartLoadRatio ) * CompOffMassFlow );
        AverageOAMassFlow = ( PartLoadRatio * OACompOnMassFlow ) + ( ( 1 - PartLoadRatio ) * OACompOffMassFlow );
    } else {
        AverageUnitMassFlow = CompOnMassFlow;
        AverageOAMassFlow = OACompOnMassFlow;

        // Added by BAN January 19, 2016
        // Average air flow when the compressor is off should be reset as follows:
        if ( PartLoadRatio == 0.0 ) {
            AverageUnitMassFlow = CompOffMassFlow;
            AverageOAMassFlow = OACompOffMassFlow;
        }
    }
```

Also I see in several places OA mass flow rate when compressor off was set to OA volume flow rate
OACompOffMassFlow = VRFTU( VRFTUNum ).NoCoolHeatOutAirVolFlow;

This will be corrected as follows:
OACompOffMassFlow = VRFTU( VRFTUNum ).NoCoolHeatOutAirMassFlow;

These two defects will be addressed in this bug issue:

**Comment by Nigusse** [comment]
The initial fix resulted diffs in the new VRF Model because the CompOnMassFlow variable is reset to  "OACompOnMassFlow" variable when the compressor PLR is zero. This logic in not used in the original VRF model.  For now the fix will only reset to AverageOAMassFlow as follows.

 // set the average OA air flow to off compressor values if the compressor PartLoadRatio is zero
      if ( PartLoadRatio == 0.0 ) {
            AverageOAMassFlow = OACompOffMassFlow;
        }

**Comment by rraustad** [comment]
This change looks on the right track. The CompOff flow rates for supply air and outdoor air need to be included when setting flow rates. Care should be taken to allow the new VRF model to do whatever it was programmed to do until we determine the new model should be revised.

**Comment by Myoldmopar** [comment]
Closed via #5431
